resources/repo.rb

```diff
@@ -2,6 +2,7 @@
 property :repo_type, String, default: 'maven2-hosted'
 property :attributes, Hash, coerce: ::Nexus3::Helper.method(:coerce_repo_attributes), default: lazy { ::Mash.new }
 property :online, [true, false], default: true
+property :routing_rule_name, String, default: ''
```

Comment:
Accept both `String` and `NilClass` to handle/simplify the case where no routing rule has been set on a repo.

Reply:
When there is no rule set, default value is empty string. Also if we have like here default: '', then if not set it would have been changed into a string too, no ?

Reply:
> Also if we have like here default: '', then if not set it would have been changed into a string too, no ?
True for repositories created using this resource, but already-existing one or created manually could have a different default value.

Reply:
As discussed, tested and what you found, if nil is passed, getter is called, which means it uses the default value.